packages/fxa-payments-server/src/lib/apiClient.test.ts

```diff
@@ -422,6 +422,28 @@ describe('API requests', () => {
       });
       requestMock.done();
     });
+
+    it(`POST {auth-server}${path} with coupon`, async () => {
+      const expected = { sourceCountry: 'US', subscription: { what: 'ever' } };
+      const requestMock = nock(AUTH_BASE_URL).post(path).reply(200, expected);
+
+      const paramsWithPromo = Object.assign(params, { promoCode: 'TEST' });
+
+      expect(
+        await apiCreateSubscriptionWithPaymentMethod(paramsWithPromo)
+      ).toEqual(expected.subscription);
+
+      expect(
+        <jest.Mock>createSubscriptionWithPaymentMethod_PENDING
+      ).toBeCalledWith(metricsOptions);
+      expect(
+        <jest.Mock>createSubscriptionWithPaymentMethod_FULFILLED
+      ).toBeCalledWith({
+        ...metricsOptions,
+        sourceCountry: expected.sourceCountry,
+      });
```

Comment:
I see that you've added the promotion code to the event properties, so I'd expect it to be in the metrics options for these assertions.

Reply:
added